### DISCUSSION TITLE
How to disable Zed from adding semicolons to javascript on save

### DISCUSSION THREAD

**Discussion by drale2k** [zed-industries/zed#10446] [Feature Requests]
I would like to disable Zed from automatically adding semicolons on save. Ideally even have Zed remove them. 

I looked up [language overrides](https://docs.zed.dev/configuration/configuring-zed#language-overrides) but it seems not supported there. Not sure where to look.

In VSCode there a setting for this `"javascript.format.semicolons": "remove"`. Is there a way to do this with Zed?

**Comment by SomeoneToIgnore** [comment]
I believe what you see is the "format on save" feature in play, in particular, its default `prettier` fallback to JavaScript.
You could either use `prettier` settings in Zed (or configure your project to use prettier and configure it regularly) to enable it: https://prettier.io/docs/en/options.html#semicolons or disable the formatting on save: https://github.com/zed-industries/zed/issues/10353#issuecomment-2050411826

**Reply by BuonOmo** [reply]
I think what OP want (at least, it is what I want!) is to keep using `typescript-language-server` with the ability to add [format configurations](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md#workspacedidchangeconfiguration).

**Comment by qualityshepherd** [comment]
Okay, this is absolutely a deal breaker. If I can ditch Prettier (which sucks ass) and instead use Standard, I'm in.

**Reply by deadcoder0904** [reply]
Yes, annoying as hell.

**Comment by YO-SC** [comment]
Hey man, I followed some advice I read here + on the docs. I could achieve the desired result by creating a `.prettierrc.json` file on the root of my project. I then told Zed to use prettier via the configs. Cheers

**.prettierrc.json**
```json
{
  "singleQuote": true,
}
```

**settings.json (zed)**
```json
{
...
"languages": {
    "JavaScript": {
      "formatter": {
        "external": {
          "command": "prettier",
          "arguments": ["--stdin-filepath", "{buffer_path}"]
        }
      }
    },
    "TypeScript": {
      "formatter": {
        "external": {
          "command": "prettier",
          "arguments": ["--stdin-filepath", "{buffer_path}"]
        }
      }
    }
  },
}
```

Works with `format on save`, at least for me. Currently using Zed `v0.181.8`

**Sources:**
https://prettier.io/docs/configuration
https://json.schemastore.org/prettierrc
https://zed.dev/docs/languages/javascript?highlight=prettier#code-formatting

(I would too like to use the native language servers, but oh well.)

**Comment by TTSKarlsson** [comment]
I tried zed and this was my first hurdle. I can't just open a sporadic javascript or typescript file from terminal without zed ruining it with random unnecessary semicolons. Now back to better pastures.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by lefamoffat** [comment]
For someone who's mainly using the web to debug issues, moving something like this to e.g. Discord would be a disservice.

Having zed automatically add unwanted code changes, e.g. semicolons at the end of lines, is a definite deal breaker and just prevents people from swapping their existing IDE setups with Zed.

**Comment by admiralnelson** [comment]
anyone know how to fix this? no i do not want to install pretier

**Comment by TTSKarlsson** [comment]
They defiinitely felt threatened from having an editor not supporting javascript/typescript so they moved the conversation behind a bad actor wall.
